WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/DesignPreviewFragment.kt

```diff
@@ -15,22 +12,17 @@ import javax.inject.Inject
  */
 class DesignPreviewFragment : LayoutPreviewFragment() {
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
-    private lateinit var viewModel: HomePagePickerViewModel
 
     companion object {
         const val DESIGN_PREVIEW_TAG = "DESIGN_PREVIEW_TAG"
 
         fun newInstance() = DesignPreviewFragment()
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
+    override fun getViewModel() =
+            ViewModelProvider(requireActivity(), viewModelFactory).get(HomePagePickerViewModel::class.java)
 
-        chooseButton.setText(R.string.hpp_choose_button)
-
-        viewModel = ViewModelProvider(requireActivity(), viewModelFactory).get(HomePagePickerViewModel::class.java)
-        setViewModel(viewModel)
-    }
+    override fun getChooseButtonText() = R.string.hpp_choose_button
```

Comment:
Praise (❤️): Great simplification here, kudos!
Suggestion (💡): Actually, I would consider simplifying all this even more, this class can end up looking like this in the end:
```
class DesignPreviewFragment : LayoutPreviewFragment() {
    companion object {
        const val DESIGN_PREVIEW_TAG = "DESIGN_PREVIEW_TAG"

        fun newInstance() = DesignPreviewFragment()
    }

    override fun getChooseButtonText() = R.string.hpp_choose_button

    override fun getViewModel() =
            ViewModelProvider(requireActivity(), viewModelFactory).get(HomePagePickerViewModel::class.java)
}
```

To break things down:
1) `@Inject lateinit var viewModelFactory: ViewModelProvider.Factory` and the `onAttach(...)` function can be moved to the parent `LayoutPreviewFragment` class.
2) On the `AppComponent` level, the `inject(...)` function for the child `DesignPreviewFragment` class can be replaced with the parent `LayoutPreviewFragment` class.
3) And that's it, the parent class will be solely responsible for the injection mechanism using the child classes the direct towards the right `ViewModel` instance. I actually tested it and it works as expected.

PS: Same applies for the child `BlockLayoutPreviewFragment` class.

Reply:
@ParaskP7 - sure 👍

Reply:
Updated in 2e428c2